tests/third_party/osmosis_testnet/net_config.py

```diff
@@ -18,42 +18,40 @@
 # ------------------------------------------------------------------------------
 
 """Osmosis network config."""
-from time import sleep
-
-import requests
-
+from cosmpy.aerial.client import LedgerClient
 from cosmpy.aerial.config import NetworkConfig
+from cosmpy.aerial.wallet import LocalWallet
+
 
+# Network config
+PREFIX = "osmo"
+DENOM = "uosmo"
+GAS_LIMIT = 120000
 
 NET_CONFIG = NetworkConfig(
     chain_id="osmo-test-4",
     url="grpc+http://grpc-test.osmosis.zone:443/",
     fee_minimum_gas_price=1,
-    fee_denomination="uosmo",
-    staking_denomination="uosmo",
+    fee_denomination=DENOM,
+    staking_denomination=DENOM,
 )
+
 DEFAULT_TIMEOUT = 60.0
+FUNDED_PK = "974c467d48351204b8481736079edfefa98eba16f4b80d74b279e103e8ee9246"
 
 
 class FaucetMixIn:
     """Osmosis faucet config"""
 
-    def ask_funds(self, wallet):
+    def ask_funds(self, wallet: LocalWallet, ledger: LedgerClient):
         """Request fund from faucet.
 
         :param wallet: Wallet Address
+        :param ledger: LedgerClient
         :raises Exception: fail to topup
         """
-        resp = requests.post(
-            "https://testnet-faucet.dev-osmosis.zone/request",
-            json={"address": str(wallet.address())},
-            timeout=DEFAULT_TIMEOUT,
-        )
-        assert resp.status_code == 200
-        ledger = self.get_ledger()
-        for i in range(10):
-            if ledger.query_bank_balance(wallet.address()) > 0:
-                break
-            sleep(i * 2)
-        else:
-            raise Exception("fail to topup")
+
+        funded_wallet = LocalWallet.from_unsafe_seed(FUNDED_PK, prefix=PREFIX)
+        ledger.send_tokens(
+            wallet.address(), 100, DENOM, funded_wallet
```

Comment:
Just wondering how much funds does the `FUNDED_PK` have? In any case, we could probably send the minimum amount here, to maximize the amount of tests we can do from the funds available. Or another possibility would be to send back and forth between two funded wallets.

Reply:
is there any possibility to make these osmoses tests independent from external testnet?
For example by creating a docker setup (e.g. via docker compose thing, or similar approach) which can be started as local node and would substitute external testnet. That way, these tests will be self-sufficient = can be run independently from external testnet.

Reply:
We can do it in next PR.

Currently tests are optional and funded account gets Coin(amount='100000000', denom='uosmo')